client/app/routes/create-account.tsx

```diff
@@ -22,7 +35,24 @@ export const action = async ({ request }: ActionFunctionArgs) => {
 
 	const { password, email, username } = await request.json();
 
-	const user = await prisma.user.create({ data: { email, username } });
+	let user: User;
+	try {
+		user = await prisma.user.create({ data: { email, username } });
+	} catch (error) {
+		if (error instanceof Prisma.PrismaClientKnownRequestError) {
+			const [field] = error.meta?.target as string[];
+			return json(
+				{ type: "conflict", field, message: `${field} already exists` },
+				{ status: 403 },
+			);
+		}
+		return json(
+			{ type: "db-error", message: "something wrong happened" },
+			{ status: 500 },
+		);
+	}
+
+	user = await prisma.user.create({ data: { email, username } });
```

Comment:
Does providing a unique email/user not cause a conflict still?

We’re creating the user twice. See line 40. The user is created there too.